lib/rex/random_identifier/generator.rb

```diff
@@ -34,10 +34,40 @@ class ExhaustedSpaceError < StandardError; end
     # This should be pretty universal for identifier rules
     :char_set => Rex::Text::AlphaNumeric+"_",
     :first_char_set => Rex::Text::LowerAlpha,
-    :forbidden => []
+    :forbidden => [].freeze
+  }
+
+  JavaOpts = DefaultOpts.merge(
+    forbidden: (
+      DefaultOpts[:forbidden] +
+      %w[
+        abstract assert boolean break byte case catch char class const
+        continue default do double else enum extends false final finally
+        float for goto if implements import instanceof int interface long
+        native new null package private protected public return short
+        static strictfp super switch synchronized this throw throws
+        transient true try void volatile while
+      ]
+    ).uniq.freeze
+  )
+
+  JSPOpts = JavaOpts.merge(
+    forbidden: (
+      JavaOpts[:forbidden] +
+      %w[
+        _ application config out page pageContext request response session var
+      ]
+    ).uniq.freeze
+  )
+
+  Opts = {
+    default: DefaultOpts,
+    java: JavaOpts,
+    jsp: JSPOpts
   }
```

Comment:
In the future this could most likely have powershell etc too, but for now we're just dealin with Java/JSP